include/picongpu/particles/collision/detail/ListEntry.hpp

```diff
@@ -28,273 +28,368 @@
 #    include <mallocMC/mallocMC.hpp>
 #endif
 
-namespace picongpu
+namespace picongpu::particles::collision
 {
-    namespace particles
+    namespace detail
     {
-        namespace collision
+        /** Index access to particles within a cell of an supercell.
+         *
+         * An accessor provides access to all particles within a cell (NOT supercell).
+         *
+         * @tparam T_FramePtrType frame pointer type
+         */
+        template<typename T_FramePtrType>
+        struct ParticleAccessor
         {
-            namespace detail
+            T_FramePtrType* m_framePtrList = nullptr;
+            uint32_t* m_parIdxList = nullptr;
+            uint32_t m_numPars = 0u;
+
+            static constexpr uint32_t frameSize = T_FramePtrType::type::frameSize;
+
+            DINLINE ParticleAccessor(uint32_t* parIdxList, uint32_t const numParticles, T_FramePtrType* framePtrList)
+                : m_framePtrList(framePtrList)
+                , m_parIdxList(parIdxList)
+                , m_numPars(numParticles)
             {
-                /* Storage for particle IDs.
-                 *
-                 * Storage for IDs of collision eligible macro particles. It comes with a simple
-                 * shuffling algorithm.
-                 */
-                struct ListEntry
-                {
-                    //! Size of the particle list (number of stored IDs).
-                    uint32_t size;
-                    //! Pointer to the actual data stored on the device heap.
-                    uint32_t* ptrToIndicies;
-
-                    /* Initialize storage, allocate memory.
-                     *
-                     * @param acc alpaka accelerator
-                     * @param deviceHeapHandle Heap handle used for allocating storage on device.
-                     * @param numPar maximal number of IDs to be stored in the list.
-                     */
-                    template<typename T_Worker, typename T_DeviceHeapHandle>
-                    DINLINE void init(T_Worker const& worker, T_DeviceHeapHandle& deviceHeapHandle, uint32_t numPar)
+            }
+
+            /** Number of particles within the cell
+             */
+            DINLINE uint32_t size() const
+            {
+                return m_numPars;
+            }
+
+            /** get particle
+             *
+             * @param idx index of the particle, range [0;size())
+             */
+            DINLINE auto operator[](uint32_t idx) const
+            {
+                return m_framePtrList[idx / frameSize][idx % frameSize];
```

Comment:
```suggestion
                const uint32_t inSuperCellIdx = m_parIdxList[idx];
                return m_framePtrList[inSuperCellIdx / frameSize][inSuperCellIdx % frameSize];
```
If i understand ithe code correctly, this is most likely the fix to our problem.  I will check if it helps.

Reply:
With this fix it looks good again, but I still need to rerun all tests in full.

Reply:
nice catch!